src/sources/http_scrape/integration_tests.rs

```diff
@@ -0,0 +1,273 @@
+//! Integration tests for http_scrape source.
```

Comment:
Random thought/nit about organization of test cases - we should determine a standard organization/naming scheme. Just to keep things easier to find.

Ex: successes then failures, all failures named `error_*`, etc

Not related or intended for this PR but came to mind as I was reviewing - especially as we take more ownership of integrations.